lib/resource_registry/relationship_type.rb

```diff
@@ -127,5 +130,22 @@ def valid_relationship_field?(read_dto)
     sig { overridable.returns(T.nilable(Symbol)) }
     def relationship_field_name
     end
+
+    private
+
+    sig { void }
+    def validate_spec!
```

Comment:
Shouldn't we have a class that defines all these, and raise these errors when instantiating it?

Reply:
I thought about it but it seemed overkill for now.